subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/logging/GroupedTaskFixture.java

```diff
@@ -53,7 +56,13 @@ public String getName() {
     }
 
     public String getOutput() {
-        return CollectionUtils.join("\n", outputs);
+        List<String> nonEmptyOutputs = filter(outputs, new Spec<String>() {
```

Comment:
Because `outputs` contains items not only for output, but also for when the task header is printed without any output.  If the task header is printed without output, it shows up in `outputs` as an empty string.  When we turn this output list into a string, we join it with `\n` which effectively adds output that the task never produced (i.e. new lines).  This filter is just to prevent those empty strings from ending up as new lines in the output string.

There are other ways to fix this of course, but this just seemed like the least intrusive.